src/distributions/mod.rs

```diff
@@ -77,7 +77,7 @@ impl<Sup> RandSample<Sup> {
 /// A value with a particular weight for use with `WeightedChoice`.
 pub struct Weighted<T> {
     /// The numerical weight of this item
-    pub weight: usize,
+    pub weight: u64,
```

Comment:
Yes, I think `u64` is definitely enough precision for this purpose (when will one want to have weighted random generation where the probabilities are more precise than 2<sup>-64</sup>?), but `u32` is not clearly so for me... although 2<sup>32</sup> is pretty big too.